### PR TITLE
LPD-41430 [Upgrade Process] Delete notification entities belonging to a non-existent company

### PR DESCRIPTION
**Pull request by p-albuquerque**
Related issue: https://issues.liferay.com/browse/LPD-41430

### CODE REVIEW

**Comment by p-albuquerque** [comment]
ci:test:sf

**Comment by marcelabc** [comment]
Just started reviewing :)

**Review by marcelabc — CHANGES_REQUESTED** [review]
Hi 
I finished my review. Let me know if you have questions :smile:

**Comment by p-albuquerque** [comment]
ci:test:sf

**Comment by p-albuquerque** [comment]
ci:test:relevant

File: modules/apps/notification/notification-service/src/main/java/com/liferay/notification/internal/upgrade/v3_10_4/DeleteStaleNotificationUpgradeProcess.java [hunk 1]
```diff
@@ -0,0 +1,244 @@
...
+	}
+
+	private void _deleteResourcePermissions(
+			String className, long primarykey, long companyId)
+		throws Exception {
+
+		_resourcePermissionLocalService.deleteResourcePermissions(
+			companyId, className, ResourceConstants.SCOPE_INDIVIDUAL,
+			String.valueOf(primarykey));
+	}
```
Comment (marcelabc) [inline modules/apps/notification/notification-service/src/main/java/com/liferay/notification/internal/upgrade/v3_10_4/DeleteStaleNotificationUpgradeProcess.java:189]: Sort the parameters following the order in `_resourcePermissionLocalService.deleteResourcePermissions()`:
`companyId > className > primarykey`

File: modules/apps/notification/notification-service/src/main/java/com/liferay/notification/internal/upgrade/v3_10_4/DeleteStaleNotificationUpgradeProcess.java [hunk 2]
```diff
@@ -0,0 +1,244 @@
...
+
+	private PreparedStatement _getDeletePreparedStatement(
+			String tableName, String dbColumnName)
+		throws Exception {
```
Comment (marcelabc) [inline modules/apps/notification/notification-service/src/main/java/com/liferay/notification/internal/upgrade/v3_10_4/DeleteStaleNotificationUpgradeProcess.java:193]: sort please

Comment (marcelabc) [reply modules/apps/notification/notification-service/src/main/java/com/liferay/notification/internal/upgrade/v3_10_4/DeleteStaleNotificationUpgradeProcess.java:193]: Why not just `columnName`? Can you rename it?

File: modules/apps/notification/notification-service/src/main/java/com/liferay/notification/internal/upgrade/v3_10_4/DeleteStaleNotificationUpgradeProcess.java [hunk 3]
```diff
@@ -0,0 +1,244 @@
...
+				"notificationRecipientId = ", notificationRecipientId));
+	}
+
+	private void _deleteNotificationTemplate() throws Exception {
+		try (PreparedStatement deletePreparedStatement1 =
+				 _getDeletePreparedStatement("NotificationTemplate",
+					 "notificationTemplateId");
+			 PreparedStatement deletePreparedStatement2 =
+				 _getDeletePreparedStatement("NTemplateAttachment",
+					 "notificationTemplateId");
```
Comment (marcelabc) [inline modules/apps/notification/notification-service/src/main/java/com/liferay/notification/internal/upgrade/v3_10_4/DeleteStaleNotificationUpgradeProcess.java:141]: Can you sort it, please?
**N**TemplateAttachment > **No**tification**T**emplate
`N` should come first, then `No`.
Right?

File: modules/apps/notification/notification-test/build.gradle [hunk 4]
```diff
@@ -15,5 +15,7 @@ dependencies {
 	testIntegrationImplementation project(":apps:object:object-rest-api")
 	testIntegrationImplementation project(":apps:object:object-test-util")
 	testIntegrationImplementation project(":apps:portal-vulcan:portal-vulcan-api")
+	testIntegrationImplementation project(":apps:portal:portal-upgrade-test-util")
+	testIntegrationImplementation project(":apps:static:portal:portal-upgrade-api")
```
Comment (marcelabc) [inline modules/apps/notification/notification-test/build.gradle:19]: Can you sort it?
```
testIntegrationImplementation project(":apps:portal:portal-upgrade-test-util")
testIntegrationImplementation project(":apps:portal-vulcan:portal-vulcan-api")
testIntegrationImplementation project(":apps:static:portal:portal-upgrade-api")
```

File: modules/apps/notification/notification-service/src/main/java/com/liferay/notification/internal/upgrade/v3_10_4/DeleteStaleNotificationUpgradeProcess.java [hunk 5]
```diff
@@ -0,0 +1,244 @@
...
+import com.liferay.portal.kernel.util.PortalUtil;
+
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+
+/**
+ * @author Paulo Albuquerque
+ */
+public class DeleteStaleNotificationUpgradeProcess extends UpgradeProcess {
```
Comment (marcelabc) [inline modules/apps/notification/notification-service/src/main/java/com/liferay/notification/internal/upgrade/v3_10_4/DeleteStaleNotificationUpgradeProcess.java:36]: I searched in the portal and found that DeleteStaleSomething is used in the plural, like `deleteStaleDocuments()`.
Can you rename it to `DeleteStaleNotificationQueueEntriesAndNotificationTemplatesUpgradeProcess`?

File: modules/apps/notification/notification-test/src/testIntegration/java/com/liferay/notification/internal/upgrade/v3_10_4/test/DeleteStaleNotificationUpgradeProcessTest.java [hunk 6]
```diff
@@ -0,0 +1,360 @@
...
+import org.junit.ClassRule;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+/**
+ * @author Paulo Albuquerque
+ */
+@RunWith(Arquillian.class)
+public class DeleteStaleNotificationUpgradeProcessTest {
```
Comment (marcelabc) [inline modules/apps/notification/notification-test/src/testIntegration/java/com/liferay/notification/internal/upgrade/v3_10_4/test/DeleteStaleNotificationUpgradeProcessTest.java:46]: Can you rename the test following the previous suggestion?

**Review by marcelabc — CHANGES_REQUESTED** [review]
Hi 
I finished my review. Let me know if you have questions :smile:

**Comment by p-albuquerque** [comment]
ci:test:sf

**Review by marcelabc — APPROVED** [review]
LGTM 🎉

**Comment by marcelabc** [comment]
ci:test:relevant

**Comment by marcelabc** [comment]
ci:test:object

**Comment by marcelabc** [comment]
Hey @p-albuquerque 
Even though the stable has failed (a known issue), you can check the results using the Testray link.
You can manually forward it to Brian if there are no unique failures.
Thank you!

**Comment by p-albuquerque** [comment]
@marcelabc Okay, I'm starting the analysis now

**Comment by p-albuquerque** [comment]
ci:test:sf

**Comment by p-albuquerque** [comment]
ci:test:relevant

**Comment by marcelabc** [comment]
ci:test:relevant

**Comment by marcelabc** [comment]
ci:report:87558240

**Comment by marcelabc** [comment]
ci:forward:force

**Pull request closed without merging**